# Hybrid-ML-Prediction
# Hybrid Machine Learning Model for smORFs Prediction in _Mycobacterium tuberculosis_

This repository contains supplementary materials and references for the manuscript **"Development and Evaluation of a Hybrid Machine Learning Algorithm for smORFs Prediction in _Mycobacterium tuberculosis_: A Comparison with Standalone Models."**

## Repository Structure

- **Supplementary Materials/**: Additional files supporting the manuscript.
  - **Mtb_positive_negative_data.csv**: Cleaned dataset for training and testing.
  - **Predicted_46_smORFs/**: Files related to predicted small open reading frames (smORFs):
    - **Supplementary File A**: Predicted smORFs in FASTA format.
    - **Supplementary File B**: BLAST results for transcript comparison with predicted mtb_smORFs.
    - **Supplementary File C**: Overlap analysis with pathogenic and non-pathogenic bacteria with mtb_smORFs.
    - **Supplementary File D**: Comparison of mtb_smORFs with _Trichophyton rubrum_ data.
    - **Supplementary File**: Comparison of predicted smORFs with previous studies small proteins.
    - **Supplementary File F**: Output from Blast2GO functional annotation.
    - **Supplementary File G**: 16S rRNA sequences from 46 bacterial species.
    - **Supplementary File H**: Performance metrics for smORFs across 46 bacterial genomes.
    - **Supplementary Table 1**: List of bacterial genomes used in the study.
    - **Supplementary Table 2**: Transcriptome dataset ID for M. tuberculosis.
    - **Supplementary Table 3**: _Trichophyton rubrum_ fungi dataset.
    - **Supplementary Table 4**: Functional annotation of predicted smORFs.

- **hybrid_mtb_prediction.ipynb**: Jupyter Notebook for the hybrid model implementation.

## References

Cited references are included in the supplementary materials.
